unblob/handlers/archive/arc.py

```diff
@@ -40,12 +42,19 @@ class ARCHandler(StructHandler):
         ushort date;	/* creation date */
         ushort time;	/* creation time */
         short crc;	/* cyclic redundancy check */
-        long length;	/* true file length */
+        ulong length;	/* true file length */
     };
     """
 
     HEADER_STRUCT = "heads"
 
+    def valid_name(self, name: bytes) -> bool:
+        try:
+            snull(name.decode("utf-8"))
```

Comment:
We first want to strip nulls, _then_ convert to string:
```suggestion
            snull(name).decode("utf-8")
```
This'll fix pyright and the test failure too